Add ReleaseMetadataFileName to mod pack item

jroweboy opened the pull request:
Previously if a repo uses the multirelease scheme, then the test mod pack feature would fail due to defaulting to the RMFN for the default repo. This change adds this field onto the mod pack item and uses that for configuring the download location.

Reproduction steps for the bug:

Setup a github release with custom release metadata filename
Configure mod to use this release filename
Add mod to mod pack
Use the test mod pack button.

Result: A warning pops up claiming that update support isn't setup correctly or a release is missing because Reloaded-II is using `Sewer56.Update.ReleaseMetadata.json` by default.

Sewer56 reviewed (changes requested):
Looks good to me.

Only thing to note; this PR changes the signature of `TryDownloadAsync` which breaks backwards compatibility. This is not a big deal; but there is possibility in the future UpdaterData might contain more context info in which case it shouldn't be changed. Please do revert that part.

jroweboy commented:
Okay, I reverted that part and changed it to update the Metadata name if its different for that item. https://github.com/Reloaded-Project/Reloaded-II/compare/75cd489f8036fe6163cd4828fdc62c2fafbcdd23..8f2f0a097540c7defa3cbeb7e5d84908409d4d0e#diff-8d6029e854f30b04c24caa47e4c9758c53fc998c45abc2fdfa9d05d5150693a2R55-R58

Sewer56 reviewed (approved):
Looks Good To Me
